proxy/session.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+type session struct {
+	IsActive    bool
+	MediatorURL string `json:"mediatorUrl"`
+}
+
+func sessionHandler(currentSession *session) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path == "/session" {
```

Comment:
It is considering a good practice to keep the green path in the root of the method. Something like:
```
if req.URL.Path != "/session" {
    return 
}

if !(req.Mathod == "POST" || req.Mathod == "DELETE") {
    return
}

...
```

This will also enable you to add proper error response codes in those cases.

Reply:
Weren't empty returns a bad practice?

Reply:
Also I think that the req.Methods don't need to be done like that, because I'll have to check for them later anyway, so I left it like:

`if req.URL.Path != "/session" {
        return
     }
     if req.Method == "POST" {
       startSession(currentSession, req)
     }
     if req.Method == "DELETE" {
       currentSession.IsActive = false
     }
`